src/cascadia/TerminalCore/TerminalSelection.cpp

```diff
@@ -307,12 +307,14 @@ std::pair<til::point, til::point> Terminal::_ExpandSelectionAnchors(std::pair<ti
         // GH#5099: We round to the nearest cell boundary,
         //   so we would normally prematurely expand to the next word
         //   as we approach it during a 2x-click+drag.
-        // - !IsWordBoundary(): fixes issue above by prohibiting expanding to the next word if we're at the word boundary
-        // - start == end: allow word expansion when 2x-clicking word boundary
-        if (start == end || !buffer.IsWordBoundary(end, _wordDelimiters))
+        //   To remedy this, decrement the end's position by 1.
+        //   However, only do this when expanding right (it's correct
+        //   as is when expanding left).

```

Comment:
I'm not 100% sure if I understand this code, but to me it feels like the calling code should contain the fix for this and not this function. Is this perhaps just a case of `floor` VS. `ceil` in the calling code?

Reply:
Yeah, I'm not happy about this part either. Let me explain what's going on better. The code flow is:
- `ControlInteractivity::SetEndSelectionPoint()` ([link](https://github.com/microsoft/terminal/blob/bee9371c0f00d36ec206bce740f6a614e6a1e06e/src/cascadia/TerminalControl/ControlInteractivity.cpp#L673)) - called on mouse drag
   - `ControlInteractivity::_getTerminalPosition()` ([link](https://github.com/microsoft/terminal/blob/bee9371c0f00d36ec206bce740f6a614e6a1e06e/src/cascadia/TerminalControl/ControlInteractivity.cpp#L688)): round to the nearest cell, that way, we have to drag past a cell to "select" it
- `ControlCore::SetEndSelectionPoint()` ([link](https://github.com/microsoft/terminal/blob/bee9371c0f00d36ec206bce740f6a614e6a1e06e/src/cascadia/TerminalControl/ControlCore.cpp#L1211)): really just clamps the position to the viewport bounds
- `Terminal::SetSelectionEnd()` ([link](https://github.com/microsoft/terminal/blob/bee9371c0f00d36ec206bce740f6a614e6a1e06e/src/cascadia/TerminalCore/TerminalSelection.cpp#L205)) --> `Terminal::_SetSelectionEnd()` ([link](https://github.com/microsoft/terminal/blob/bee9371c0f00d36ec206bce740f6a614e6a1e06e/src/cascadia/TerminalCore/TerminalSelection.cpp#L216)):
   - This is the important part of the code, really. It's where we update our selection state for shift+click, pivoting, and multi-clicks (+ dragging). 
   - The issue really only arises when we're multi-clicking+dragging (think "double-click" then drag, as it should expand by word as we drag). So the issue has to do with the expansion part. Anything before that in this function wouldn't make sense to alter, imo.
- `Terminal::_ExpandSelectionAnchors()` ([link](https://github.com/microsoft/terminal/blob/bee9371c0f00d36ec206bce740f6a614e6a1e06e/src/cascadia/TerminalCore/TerminalSelection.cpp#L290C46-L290C68)): this is where we actually expand our selection endpoints if we're multi-clicking. 

Thankfully, `_ExpandSelectionAnchors()` is only used in this spot. I'm not sure where else the change would make sense to go though. Open to suggestions.